### PR TITLE
[core] [1/N] New interface to combine client registerat and announce at NodeManager

### PR DESCRIPTION
**Pull request by dentiny**
Issue reference: https://github.com/ray-project/ray/issues/48837
This PR simply combines two existing API call: client registration and port announcement into one, with no functionality change.

The plan is to
- Add new APIs at callee side, so we don't have production effect;
- Do no-op code refactor at caller side, so it's easier to refactor and merge API calls
- Connect caller and callee with the new API calls, after (1) and (2) the code structure should be at a clean state which is friendly and easy for API merge.

### CODE REVIEW

**Comment by dentiny** [comment]
Hi @jjyao can I get a review for this PR? It's been a week, thanks!

File: src/ray/raylet/format/node_manager.fbs [hunk 1]
```diff
@@ -144,6 +150,20 @@ table RegisterClientReply {
...
+}
+
+table RegisterClientWithPortResponse {
+  // Response to register client.
+  request_client_request: [RegisterClientReply];
+  // Response to assign port.
+  announcement_port_request: [AnnounceWorkerPortReply];
+}
+
 table AnnounceWorkerPort {
```
[the remark this answers is not in the document]
Comment (dentiny) [reply src/ray/raylet/format/node_manager.fbs:167]: Discussed offline, my plan is to
- Add new APIs at callee side, so we don't have production effect;
- Do no-op code refactor at caller side, so it's easier to refactor and merge API calls
- Connect caller and callee with the new API calls, after (1) and (2) the code structure should be at a clean state which is friendly and easy for API merge.

[the remark this answers is not in the document]
Comment (dentiny) [reply src/ray/raylet/format/node_manager.fbs:167]: Step (1) and (2) should have no affect to production.

**Comment by dentiny** [comment]
Hi @rynewang wondering if you have further feedbacks?

**Review by rynewang — APPROVED** [review]

**Pull request merged**